### PR TITLE
Support OIDC

### PR DESCRIPTION
**Pull request by devkiran**
## What does this PR do?

This PR introduce OIDC flow for Cal.com

Settings UI https://www.loom.com/share/6b60dbfa0b4643259fa9c99f735402da

Fixes # (issue)

**Environment**: Staging(main branch) / Production

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## How should this be tested?

- Add OIDC connection via `/settings/security/sso` or from team settings
- Come to the Login page and click on the `Sign in with SAML` button

### CODE REVIEW

**Comment by devkiran** [comment]
@deepakprabhakara Please take a look at this PR and let me know if we need to change anything.

File: apps/web/pages/api/auth/oidc.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { NextApiRequest, NextApiResponse } from "next";
```
Comment (devkiran) [inline apps/web/pages/api/auth/oidc.ts:1]: This new API route is used to get the response from the OIDC provider.

File: packages/features/ee/sso/components/ConnectionInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    });
+  };
+
+  return (
+    <div>
+      {connection.type === "saml" ? (
+        <SAMLInfo acsUrl={connection.acsUrl} entityId={connection.entityId} />
+      ) : (
+        <OIDCInfo callbackUrl={connection.callbackUrl} />
+      )}
```
Comment (devkiran) [inline packages/features/ee/sso/components/ConnectionInfo.tsx:53]: Display the Connection information based on the connection type (SAML or OIDC)

File: packages/features/ee/sso/lib/saml.ts [hunk 3]
```diff
@@ -11,6 +12,7 @@ export const samlTenantID = "Cal.com";
 export const samlProductID = "Cal.com";
 export const samlAudience = "https://saml.cal.com";
 export const samlPath = "/api/auth/saml/callback";
+export const oidcPath = "/api/auth/oidc";
```
Comment (devkiran) [inline packages/features/ee/sso/lib/saml.ts:15]: Path to receive the OIDC response.

File: packages/trpc/server/routers/viewer/sso.tsx [hunk 4]
```diff
@@ -120,4 +128,44 @@ export const samlRouter = router({
         throw new TRPCError({ code: "BAD_REQUEST", message: "Deleting SAML Connection failed." });
       }
     }),
+
+  // Update the OIDC Connection
+  updateOIDC: authedProcedure
```
Comment (devkiran) [inline packages/trpc/server/routers/viewer/sso.tsx:133]: A new method to create/update the OIDC connection.

**Comment by PeerRich** [comment]
amazing work, thank you so much for this! we'll test this and merge asap. also curious if @deepakprabhakara has any comments

**Review by alannnc — COMMENTED** [review]
Code looks good! Gonna test before approving.

**Review by zomars — APPROVED** [review]
LGTM

**Pull request merged**